src/Storage/Device.php

```diff
@@ -63,6 +88,25 @@ abstract public function getPath(string $filename, string $prefix = null): strin
      */
     abstract public function upload(string $source, string $path, int $chunk = 1, int $chunks = 1, array &$metadata = []): int;
 
+    /**
+     * Upload Data.
+     *
+     * Upload file contents to desired destination in the selected disk.
+     * return number of chunks uploaded or 0 if it fails.
+     *
+     * @param string $source
+     * @param string $path
+     * @param string $contentType
+     * @param int chunk
+     * @param int chunks
+     * @param array $metadata
+     *
+     * @throws \Exception
+     *
+     * @return int
+     */
+    abstract public function uploadData(string $data, string $path, string $contentType, int $chunk = 1, int $chunks = 1, array &$metadata = []): int;
```

Comment:
How is this different from `upload()` the naming convention is very confusing and adds complexity. Should this be a private method?

Reply:
Upload is currently expecting a file path and reading the file and uploading. However, upload data is uploading String data chunk.

Reply:
Not a blocker but I would try and change the name, any common practice in other APIs? What about uploadString?